veros/core/density/gsw.py

```diff
@@ -545,3 +545,489 @@ def gsw_dHdS(vs, sa_in, ct_in, p):
                                                                              (t4 + t6 - t242)
                                                                              - 2.0 * t237 * t200 / t247 * (t4 + t6 + t242 + t123)) / t207
     return t260
+
+"""
+==========================================================================
+  Potential density & temperature, enthropy, Gibbs free energy and its
+  derivatives from Absolute Salinity, in-situ Temperature, Sea Pressure
+  using the Thermodynamic Equation Of Seawater - 2010 (TEOS-10) package
+  (see details in IOC, 2010; http://www.teos-10.org)
+==========================================================================
+"""
+@veros_method
+def get_gsw_teos10_constants(vs, key):
+
+    constants = dict(\
+
+    db2pa = 1.0e4,\
+    rec_db2pa = 1.0e-4,\
+
+    pa2db = 1.0e-4,\
+    rec_pa2db = 1.0e4,\
+
+    pi = 3.141592653589793,\
+    deg2rad = 3.141592653589793/180.0,\
+    rad2deg = 180.0/3.141592653589793,\
+
+    gamma = 2.26e-7,\
+
+    #  cp0  =  The "specific heat" for use                         [ J/(kg K) ]
+    #          with Conservative Temperature
+    gsw_cp0 = 3991.86795711963,\
+    #  T0  =  the Celcius zero point.                                     [ K ]
+    gsw_t0 = 273.15,\
+    #  P0  =  Absolute Pressure of one standard atmosphere.              [ Pa ]
+    gsw_p0 = 101325.0,\
+    #  SSO  =  Standard Ocean Reference Salinity.                      [ g/kg ]
+    gsw_sso = 35.16504,\
+    gsw_sqrtsso = 5.930011804372737,\
+    #  uPS  =  unit conversion factor for salinities                   [ g/kg ]
+    gsw_ups = 35.16504/35.0,\
+    #  sfac  =  1/(40*gsw_ups)
+    gsw_sfac = 0.0248826675584615,\
+    #  deltaS = 24, offset = deltaS*gsw_sfac
+    offset = 5.971840214030754e-1,\
+    #  C3515  =  Conductivity at (SP=35, t_68=15, p=0)                [ mS/cm ]
+    gsw_c3515 = 42.9140,\
+    #  SonCl  =  SP to Chlorinity ratio                           [ (g/kg)^-1 ]
+    gsw_soncl = 1.80655,\
+    #  valence_factor  =  valence factor of sea salt of Reference Composition
+    #                                                              [ unitless ]
+    gsw_valence_factor = 1.2452898,\
+    #  atomic_weight = mole-weighted atomic weight of sea salt of Reference
+    #                  Composition                                    [ g/mol ]
+    gsw_atomic_weight = 31.4038218)
+
+    return constants[key]
+
+@veros_method
+def gsw_entropy_part(vs, sa, t, p):
+    """
+    Entropy minus the terms that are a function of only SA
+
+    Arguments:
+        sa     : Absolute Salinity                          [ g/kg ]
+        t      : in-situ temperature                         [deg C]
+        p      : sea pressure                                 [dbar]
+
+    Returns:
+        Entropy part
+    """
+
+    x2 = get_gsw_teos10_constants(vs, 'gsw_sfac') * sa
+    x = np.sqrt(x2)
+    y = t * 0.025
+    z = p * 1e-4
+
+    g03 = z*(-270.983805184062 + \
```

Comment:
Two formatting problems with this:

1. You are missing whitespace around operators. It should be `z * (-270...`
2. In Veros, we usually put operators at the start of the following line instead of the end of the previous line.